modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java

```diff
@@ -349,31 +338,35 @@ public void setPidDir(String pidDirStr) {
      * @since 5.4.2
      */
     public void checkPaths() throws ConfigurationException {
-        File badInstanceClid = new File(generator.getNuxeoHome(),
-                getDefaultDataDir() + File.separator + "instance.clid");
-        if (badInstanceClid.exists() && !getDataDir().equals(badInstanceClid.getParentFile())) {
-            log.warn("Moving {} to {}.", () -> badInstanceClid, this::getDataDir);
+        Path badInstanceClid = configHolder.getRuntimeHomePath()
+                                           .resolve(Environment.DEFAULT_DATA_DIR)
+                                           .resolve("instance.clid");
```

Comment:
You could have a constant `INSTANCE_CLID_PATH = Path.of(Environment.DEFAULT_DATA_DIR, "instance.clid")` and do `.resolve(INSTANCE_CLID_PATH)`

Reply:
Sure, I will name the constant `BAD_INSTANCE_CLID_PATH` to avoid confusion as the clid must follow the data path.